solutions/usaco-687.mdx

```diff
@@ -83,4 +83,63 @@ int main() {
 
 </CPPSection>
 
+<PySection>
+```py
+"""
+We're going to make a dictionary to store the milk production
+of each cow.
+Next, we're going to make a list of the dictionary
+and sort that list by milk production.
+Finally, we're going to iterate over that list
+to find who produces the second least amount of milk.
+"""
+	
+fin = open("notlast.in")
+fout = open("notlast.out", "w")
+
+n = int(fin.readline())
+cowDict = {"Bessie": 0,
+           "Elsie": 1,
+           "Daisy": 2,
+           "Gertie": 3,
+           "Annabelle": 4,
+           "Maggie": 5,
+           "Henrietta": 6}
+cowmilk = [["Bessie", 0], ["Elsie", 0], ["Daisy", 0], ["Gertie", 0],
+           ["Annabelle", 0], ["Maggie", 0], ["Henrietta", 0]]
+# We're going to use a list as a dictionary
+# Because you can't sort a dictionary
+
+for i in range(n):
+	cow, milk = fin.readline().split()
+	cowmilk[cowDict[cow]][1] += int(milk)
+
+# Sort the cows by their milk production
+cowmilk.sort(key=lambda x: x[1])
+
+# Our necessary pieces to find the second least milk
+smallest = cowmilk[0][1]
+secSmallest = None
```

Comment:
use snake_case, PEP 8 says so

Reply:
PEP 8 also says that consistency within is more important than consistency with the style guide. Since I use camelCase all throughout my code, it would be more appropriate to keep it camelCase.

Reply:
refactoring exists
i can do it for you if you want